src/components/quicklaunch.jsx

```diff
@@ -109,9 +113,46 @@ export default function QuickLaunch({
       if (searchProvider) {
         newResults.push({
           href: searchProvider.url + encodeURIComponent(searchString),
-          name: `${searchProvider.name ?? t("quicklaunch.custom")} ${t("quicklaunch.search")} `,
+          name: `${searchProvider.name ?? t("quicklaunch.custom")} ${t("quicklaunch.search")}`,
           type: "search",
         });
+
+        if (!hideSearchSuggestions && searchProvider.suggestionUrl) {
+          if (searchString.trim() !== searchSuggestions[0]) {
+            fetch(
+              `/api/searchSuggestion?query=${encodeURIComponent(searchString)}&providerName=${searchProvider.name}`,
+              { signal: abortController.signal },
+            )
+              .then(async (searchSuggestionResult) => {
+                const newSearchSuggestions = await searchSuggestionResult.json();
+
+                // Check if there is a search suggestion
+                if (newSearchSuggestions) {
+                  // Restrict the searchSuggestion to 4 entries
+                  if (newSearchSuggestions[1].length - 4 > 0) {
+                    newSearchSuggestions[1].splice(-(newSearchSuggestions[1].length - 4));
+                  }
+
+                  // Save the new search suggestions in their state.
+                  setSearchSuggestions(newSearchSuggestions);
+                }
+              })
+              .catch(() => {
+                // If there is an error, just ignore it. There just will be no search suggestions.
+              });
+          }
+
+          // Show search suggestions from their state. This will show the "old" suggestions until they are updated.
+          if (searchSuggestions[1]) {
+            newResults = newResults.concat(
+              searchSuggestions[1].map((suggestion) => ({
+                href: searchProvider.url + encodeURIComponent(suggestion),
+                name: suggestion,
```

Comment:
```suggestion
                name: `${searchProvider.name ?? t("quicklaunch.custom")} ${t("quicklaunch.search")}: ${suggestion}`,
```

Im curious what you think about this one. I think it might be better to have the name of the search in the line to make it more explicit, it's obviously visually a bit more cluttered but I think it helps when e.g. theres also a service in the list.

I think actually even better would be to add a new quick launch "type", something like "searchSuggestion" and then that could be slightly visually differentiated like smaller or gray text or something like that

Again, curious your thoughts

<img width="793" alt="Screenshot 2024-01-27 at 5 02 04 PM" src="https://github.com/gethomepage/homepage/assets/4887959/bb2dff6b-58cd-4740-a28d-9101c2719393">

Reply:
That's a good question and probably depends on personal preferences. 

I don't like how it looks with the `Google Search:` in front of the suggestions. I find it difficult to quickly see the suggestion, always having to scan the line for the colon/start of the suggestion.

I have two ideas myself down below.

![image](https://github.com/gethomepage/homepage/assets/31217036/09442d5f-70e8-473a-a79c-91c3faeaaf5e)

## Idea 1

Showing the icon of the search provider would be an easy way of indicating that it is a search and which provider is used. Ideally, the icon would be the same as the one used in the search bar. For that to work, the `ResolvedIcon` component would need to allow `react-icons` icons to be passed in as an icon. Alternatively, the search bar could use the current version of `ResolvedIcon` and the `searchProviders` object would store the SI icon string instead of the component from `react-icons`. I did not test these yet, for the images below I just hard-coded the string for the `icon` property. 

The result would look like this:

![image](https://github.com/gethomepage/homepage/assets/31217036/113f11bb-0097-4789-bed3-5f17cdc36447)

## Idea 2

My second idea would include your suggestion of creating a new "type" for the search suggestions. 
With that, we could show the suggestion part of the search query with a lower opacity or a darker color (for the example below, I used `opacity-50`).

I must say, I really like how it looks. It's super easy to see the suggestions, and it's clear which provider they use. 

![image](https://github.com/gethomepage/homepage/assets/31217036/693b1b3f-8d13-450e-9d30-7fe32b41ecc0)

Reply:
Yea, idea 2 is exactly the kind of thing I had in mind, agree it looks great.

I'm not sure about the icon but Im also sorta tempted to leave it for another PR, keep this one focused?

Reply:
I added it without the image in ed827837aa86a892fd7603e6d0dc8187138f58c2

It looks like this:
![image](https://github.com/gethomepage/homepage/assets/31217036/4f6acf0f-e52f-4d4f-8c88-a97e74af7775)